docs/configuration/protocols/segment-routing.rst

```diff
@@ -0,0 +1,564 @@
+.. _segment-routing:
+
+###############
+Segment Routing
+###############
+
+Segment Routing (SR) is a network architecture that is similar to 
+source-routing.In this architecture, the ingress router adds a list of segments,
+known as SIDs, to the packet as it enters the network. These segments represent
+different portions of the network path that the packet will take.
+
+The SR segments are portions of the network path taken by the packet, and are 
+called SIDs. At each node, the first SID of the list is read, executed as a 
+forwarding function, and may be popped to let the next node read the next SID of 
+the list. The SID list completely determines the path where the packet is 
+forwarded.
+
+Segment Routing can be applied to an existing MPLS-based data plane and defines
+a control plane network architecture. In MPLS networks, segments are encoded as
+MPLS labels and are added at the ingress router. These MPLS labels are then 
+exchanged and populated by Interior Gateway Protocols (IGPs) like IS-IS or OSPF 
+which are running on most ISPs.
+
+
+.. note:: Segment routing defines a control plane network architecture and
+  can be applied to an existing MPLS based dataplane. In the MPLS networks,
+  segments are encoded as MPLS labels and are imposed at the ingress router.
+  MPLS labels are exchanged and populated by IGPs like IS-IS.Segment Routing
+  as per RFC8667 for MPLS dataplane. It supports IPv4, IPv6 and ECMP and has
+  been tested against Cisco & Juniper routers.however,this deployment is still
+  EXPERIMENTAL for FRR .
+ 
+
+IS-IS SR Configuration
+----------------------
+
+Segment routing (SR) is used by the IGP protocols to interconnect network
+devices, below configuration shows how to enable SR on IS-IS:
+
+
+.. note:: ``Known limitations:`` 
+
+  No support for level redistribution (L1 to L2 or L2 to L1)
+
+  No support for binding SID
+
+  No support for SRLB
+
+  Only one SRGB and default SPF Algorithm is supported
+
+
+
+.. cfgcmd:: set protocols isis segment-routing global-block <high-label-value 
```

Comment:
this is not the correct syntax, same for local-block
this should be right.:

```
.. cfgcmd:: set protocols isis segment-routing global-block high-label-value <label-value>
.. cfgcmd:: set protocols isis segment-routing global-block low-label-value <label-value>
```

Reply:
done